packages/nimble/R/cppDefs_cppProject.R

```diff
@@ -228,8 +223,27 @@ cppProjectClass <- setRefClass('cppProjectClass',
                                        if(status != 0) 
                                            stop(structure(simpleError("Failed to create the shared library"),
                                                           class = c("SHLIBCreationError", "ShellError", "simpleError", "error", "condition")))
-                                       nimbleUserNamespace$sessionSpecificDll <- dyn.load(ssDllName, local = TRUE)
-                                   },                                  
+                                       return(dyn.load(ssDllName, local = TRUE))
+                                   },
+                                   compileDynamicRegistrations = function(showCompilerOutput = nimbleOptions('showCompilerOutput')) {
+                                       timeStamp <- format(Sys.time(), "%m_%d_%H_%M_%S")
+                                       dllName <- paste0("dynamicRegistrations_", timeStamp)
+                                       cppName <- paste0(dllName, ".cpp")
+                                       writeDynamicRegistrationsDotCpp(dynamicRegistrationsCppName, dynamicRegistrationsDllName)
+                                       nimbleUserNamespace$sessionSpecificDll <- compileStaticCode(dllName, cppName)
+                                   },
+                                   compileTensorflowWrapper = function(showCompilerOutput = nimbleOptions('showCompilerOutput')) {
+                                       # Load prerequisite DLLs.
+                                       library(tensorflow)
+                                       tfPath <- tf$`__path__`  # This line triggers loading of _pywrap_tensorflow_internal.so.
+                                       nimbleUserNamespace$tensorflowDll <- file.path(tfPath, 'python', '_pywrap_tensorflow_internal.so')
```

Comment:
Re `nimbleUserNamespace$tensorflowDllPath`, we will need internal access to this variable, both to check whether the library has been compiled and to add this to the linker options of `R CMD SHLIB`. I'll move it into the `nimble` namespace.